pkg/client/unversioned/request.go

```diff
@@ -427,8 +427,36 @@ func (r *Request) VersionedParams(obj runtime.Object, convertor runtime.ObjectCo
 		return r
 	}
 	for k, v := range params {
-		for _, vv := range v {
-			r.setParam(k, vv)
+		for _, value := range v {
+			// TODO: Move it to setParam method, once we get rid of
+			// FieldSelectorParam & LabelSelectorParam methods.
+			if k == unversioned.LabelSelectorQueryParam(r.apiVersion) && value == "" {
+				// Don't set an empty selector for backward compatibility.
```

Comment:
Sure, that's fine.

On Mon, Nov 30, 2015 at 11:52 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/client/unversioned/request.go
> <https://github.com/kubernetes/kubernetes/pull/17836#discussion_r46193654>
> :
>
> > @@ -427,8 +427,36 @@ func (r *Request) VersionedParams(obj runtime.Object, convertor runtime.ObjectCo
> >  		return r
> >  	}
> >  	for k, v := range params {
> > -		for _, vv := range v {
> > -			r.setParam(k, vv)
> > +		for _, value := range v {
> > +			// TODO: Move it to setParam method, once we get rid of
> > +			// FieldSelectorParam & LabelSelectorParam methods.
> > +			if k == unversioned.LabelSelectorQueryParam(r.apiVersion) && value == "" {
> > +				// Don't set an empty selector for backward compatibility.
>
> Can we do it in a separate PR?
> This PR is not introducing any regression.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17836/files#r46193654>.
>